### ISSUE TITLE
Make it multi-user 

### ISSUE THREAD

**Issue by YannBrrd** [yhat/rodeo#44]
Hi,

It would be great to have rodeo multiuser, as R Studio does ! 

Cheers,
Yann

**Comment by seandavi** [comment]
I agree with @YannBrrd.  I use RStudio server for teaching all the time and love to be able to have each student get his/her own account.

**Comment by jonrobinson2** [comment]
While I agree a more multi-user framework would be nice I think at the moment simply as a patchwork fix it would be good to also not default to a port but to be smart about which port to listen on. 

In my use case I have multiple users working on the same server but if I launch rodeo on port 5000 and don't specifically choose a port it fails. It would be nice in the interim to have smart port listening. I figure I can script it up myself but would be nice to not have it built in for other users with similar use cases.

**Comment by bbolker** [comment]
+1; I would love to teach an intro Python course with this interface (starting in January), with Rodeo running on our departmental server ... is this in the development roadmap anywhere ... ?

**Comment by glamp** [comment]
Yes I have a working version of this in beta if you'd like to try it out

**Comment by YannBrrd** [comment]
Indeed !!

Just tell me how 😊

Le dim. 8 nov. 2015 18:37, Greg Lamp <notifications@github.com> a écrit :

> Yes I have a working version of this in beta if you'd like to try it out
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/yhat/rodeo/issues/44#issuecomment-154850314>.
>
-- 
Cordialement,
Yann Barraud

**Comment by bbolker** [comment]
+1.  If you give minimal instructions/pointers about where to start I'll try to give feedback ...

**Comment by jonrobinson2** [comment]
+1 !

**Comment by edoardovivo** [comment]
+1 !!

**Comment by glamp** [comment]
great! @bbolker @jonrobinson2 @seandavi @YannBrrd will email you all individually

**Comment by jonrobinson2** [comment]
Any updates on this?

**Comment by jonrobinson2** [comment]
?

**Comment by YannBrrd** [comment]
@glamp no mail ? no instruction ? I assume you are overloaded... :-( Tell me if II can help somehow...

**Comment by bbolker** [comment]
I'm hoping that I will be able to use this for my course this coming spring (i.e., starting January 2017). If that's going to happen (which I would love), I need to be able to evaluate it/bang on it/help make it robust over the next few months ...

**Comment by TakenPilot** [comment]
Hi.  : )

A Rodeo Server version might happen in the future, but right now I'm focused on features that make it more robust and more useful as a data science tool as an ordinary Desktop app.

Multi-user functionality isn't on the near-term roadmap, as we're trying to define ourselves as different from PyCharm and Jupyter Notebook with a few better features first.

**Comment by bbolker** [comment]
Thanks for letting me know.  I think I'm going to look into https://repl.it/ ... 

I won't presume to do it, but maybe you want to close this issue now ... ?

**Comment by bbolker** [comment]
Thanks for letting me know. I think I'm going to look into https://repl.it/ ...

I won't presume to do it, but maybe you want to close this issue now ... ?
